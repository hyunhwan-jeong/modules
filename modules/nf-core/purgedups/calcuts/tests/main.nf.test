nextflow_process {

    name "Test Process PURGEDUPS_CALCUTS"
    script "../main.nf"
    process "PURGEDUPS_CALCUTS"

    tag "modules"
    tag "modules_nfcore"
    tag "purgedups"
    tag "purgedups/pbcstat"
    tag "purgedups/calcuts"

    test("test_purgedups_calcuts") {

        setup {
            run("PURGEDUPS_PBCSTAT") {
                script "../../pbcstat/main.nf"

                process {
                    """
                    input[0] = [
                        [id:'test'],// meta map
                        file(params.modules_testdata_base_path + 'genomics/sarscov2/genome/genome.paf',checkIfExists:true)
                        ]
                    """
                }
            }
        }
        when {
            params {
                module_args = ''
            }
            process {
                """
                input[0] = PURGEDUPS_PBCSTAT.out.stat
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}