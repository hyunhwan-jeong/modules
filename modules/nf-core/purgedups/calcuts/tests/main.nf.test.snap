{
    "test_purgedups_calcuts": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cutoffs:md5,e13542222f6e6e4e0d87754956641008"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.calcuts.log:md5,04053f23b4dbb7947e8b2cb820ca6681"
                    ]
                ],
                "2": [
                    "versions.yml:md5,df5b856012c4a77707e608b2a73f6dde"
                ],
                "cutoff": [
                    [
                        {
                            "id": "test"
                        },
                        "test.cutoffs:md5,e13542222f6e6e4e0d87754956641008"
                    ]
                ],
                "log": [
                    [
                        {
                            "id": "test"
                        },
                        "test.calcuts.log:md5,04053f23b4dbb7947e8b2cb820ca6681"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,df5b856012c4a77707e608b2a73f6dde"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-14T14:07:01.617419051"
    }
}